crates/autopilot/src/database/auction_transaction.rs

```diff
@@ -11,64 +9,24 @@ impl super::Postgres {
         log_index: i64,
         tx_from: H160,
         tx_nonce: i64,
-        tx_hash: H256,
     ) -> anyhow::Result<()> {
         let _timer = super::Metrics::get()
             .database_queries
             .with_label_values(&["update_settlement_tx_info"])
             .start_timer();
 
-        let mut ex = self.0.acquire().await?;
-        // This all happens in one transaction to prevent the situation where we would update
-        // settlements without updating auction_transaction.
-        ex.transaction(|ex| {
-            async move {
-                database::auction_transaction::insert_settlement_tx_info(
-                    ex,
-                    block_number,
-                    log_index,
-                    &ByteArray(tx_from.0),
-                    tx_nonce,
-                )
-                .await
-                .context("insert_settlement_tx_info")?;
-
-                // Auctions that were stored before the auction_transaction table existed need to be
-                // inserted into it.
-                // This code can be removed when all old auctions have been processed this way.
-                let auction_id = match database::auction_transaction::get_auction_id_from_tx_hash(
-                    ex,
-                    &ByteArray(tx_hash.0),
-                )
-                .await
-                .context("get_auction_id_from_tx_hash")?
-                {
-                    Some(auction_id) => auction_id,
-                    None => return Ok(()),
-                };
-                tracing::trace!(auction_id);
-
-                // If this is an auction that was stored after the auction_transaction table existed then
-                // the row has already been inserted. This should not be treated as an error.
-                // Inserting exactly the same row again as it already exist does not error in Postgres. If
-                // this wasn't the case we would need to check the the error reason is the primary key
-                // constraint to find out whether the error is benign.
-                // Because this insert does not error there is no need for the check. If our insert changed
-                // any column we would correctly error.
-                database::auction_transaction::upsert_auction_transaction(
-                    ex,
-                    auction_id,
-                    &ByteArray(tx_from.0),
-                    tx_nonce,
-                )
-                .await
-                .context("upsert_auction_transaction")?;
-
-                Ok(())
-            }
-            .boxed()
-        })
+        let mut ex = self.0.acquire().await.context("acquire")?;
+        database::auction_transaction::insert_settlement_tx_info(
```

Comment:
One question - in looking at the code for `insert_settlement_tx_info`, this is an `UPDATE` and not an `INSERT` - is this expected?

Reply:
This is expected because this table contains settlement events. When settlement events get inserted as part of our event logic they do not yet have tx_from tx_nonce set  (null). Autopilot is responsible for going through the settlement events that don't have this and updating them with the info. So this is an update.

Reply:
Oops! This updates `settlements` and not `solver_competitions` - thanks for clarifying.